From eb393a7a97b5a24db13294972787ae1688c25246 Mon Sep 17 00:00:00 2001
From: chandavath <rcrajagopal@habana.ai>
Date: Mon, 12 Feb 2024 12:16:14 +0200
Subject: [PATCH] [SW-137121] gaudi3: erom build: makefile changes

Added required files for gaudi3 erom build and moved
PHY_FW_SIZE flag to device specific makefile.

Change-Id: I27d9c871376da4a398ce53b248573ff6ccb12fe7
Signed-off-by: chandavath <rcrajagopal@habana.ai>
---

diff --git a/app/ext_rom/Makefile b/app/ext_rom/Makefile
index 34f7522..a35e518 100644
--- a/app/ext_rom/Makefile
+++ b/app/ext_rom/Makefile
@@ -8,6 +8,7 @@
 SPECS_DEV := $(shell echo $(HL_DEV) | cut -d'_' -f1)
 
 include $(HL_DEV_ARCH_COMMON)/Makefile
+include $(HL_DEV_ARCH)/Makefile
 
 CFLAGS += -I$(SPECS_DIR)
 CFLAGS += -I$(SPECS_EMBEDDED_DIR)
@@ -17,11 +18,8 @@
 CFLAGS += -I $(PREBOOT_ROOT_DIR)/include/ -I $(PREBOOT_ROOT_DIR)/ -I $(PREBOOT_ROOT_DIR)/lib/common/ -I $(PREBOOT_ROOT_DIR)/lib/
 CFLAGS += -I $(HL_DEV_ARCH)/ -I $(HL_DEV_ARCH)/../common/ -I $(SPECS_EMBEDDED_DIR)/$(BASE_DEV_NAME)/
 CFLAGS += -include $(SPECS_EMBEDDED_DIR)/$(SPECS_DEV)/$(SPECS_DEV)_reg_map.h
-CFLAGS += -include $(HL_DEV_ARCH)/dev_ext_rom.h
 CFLAGS += -include erom_app_defs.h
 CFLAGS += -I ${EMBEDDED_ROOT}/arcboot/arch
-# dummy parameter, real one required only for ppboot build
-CFLAGS += -DPHY_FW_SIZE=0
 
 CFLAGS += -Os -fdata-sections -ffunction-sections
 SFLAGS := $(CFLAGS) -D__ASSEMBLY__
@@ -38,17 +36,11 @@
 EROM_RECOVERY_APP_S := $(shell ls crt0_recovery_erom.S)
 EROM_RECOVERY_APP_OBJECTS := $(EROM_RECOVERY_APP_S:.S=.o)
 
-# TODO: Revert as a part of gaudi3 erom implementation
-ifeq ($(BASE_DEV_NAME),gaudi3)
-	EROM_PCIE_SRC := $(HL_DEV_ARCH)/erom_pcie.o
-	EROM_RECOVERY_SRC := $(HL_DEV_ARCH)/erom_recovery.o
-else
-	EROM_PCIE_SRC := erom_pcie.o
-	EROM_RECOVERY_SRC := erom_recovery.o
-	LDFLAGS += -Wl,--just-symbols $(HL_DEV_ARCH)/exports.elf
-endif
+EROM_PCIE_SRC := erom_pcie.o
+EROM_RECOVERY_SRC := erom_recovery.o
+LDFLAGS += -Wl,--just-symbols $(HL_DEV_ARCH)/exports.elf
 
-EROM_PCIE_DEPS := \
+EROM_PCIE_DEPS += \
 	$(PREBOOT_ROOT_DIR)/lib/common/pll.o \
 	$(PREBOOT_ROOT_DIR)/lib/common/pci.o \
 	$(PREBOOT_ROOT_DIR)/lib/common/uart.o \
diff --git a/arch/gaudi2/arc/Makefile b/arch/gaudi2/arc/Makefile
index 8b5bc36..7650e53 100644
--- a/arch/gaudi2/arc/Makefile
+++ b/arch/gaudi2/arc/Makefile
@@ -5,4 +5,6 @@
 
 LDFLAGS += -Wl,--just-symbols $(HL_DEV_ARCH)/exports.elf
 
+CFLAGS += -DPHY_FW_SIZE=$(PHY_FW_SIZE)
+
 image_validation: image_validation_common
diff --git a/arch/gaudi3/arc/Makefile b/arch/gaudi3/arc/Makefile
index 5e3d172..35aa7a1 100644
--- a/arch/gaudi3/arc/Makefile
+++ b/arch/gaudi3/arc/Makefile
@@ -17,4 +17,9 @@
 
 PPBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/mbist.c)
 
+EROM_PCIE_DEPS := \
+	$(PREBOOT_ROOT_DIR)/lib/common/boot_desc.o \
+	$(PREBOOT_ROOT_DIR)/lib/common/efuse.o \
+	$(PREBOOT_ROOT_DIR)/lib/common/string.o
+
 image_validation: image_validation_common
